pkg/sql/check.go

```diff
@@ -465,6 +465,47 @@ func (p *planner) RevalidateUniqueConstraint(
 	return errors.Newf("unique constraint %s does not exist", constraintName)
 }
 
+// IsConstraintActive returns if a given constraint is currently active,
+// for the current transaction.
+func (p *planner) IsConstraintActive(
```

Comment:
Is this about synthetic descriptors? I see how some of the descriptor mucking is going to hurt with the declarative schema changer.